ls1intum/Artemis
# Fix duplicate participation websocket issue in code-editor (lazy loading issue)

### Checklist

### Motivation and Context



### Description


Regression from https://github.com/ls1intum/Artemis/pull/736.
When the user enters the code-editor, there are now 2 participation-websocket services, even though it should be a singleton.
The issue is that the code-editor is lazy loaded and creates its own instance of the participation-websocket service.

There are no visible bugs atm, but the result is that every result message is emitted twice through the websocket subscription (this could e.g. lead to duplicated REST calls that are executed on reaction of a websocket message).

### Steps for Testing


1. Log in to Artemis
2. Go to the code-editor of a programming-exercise
3. Reload the page with open dev tools, check the WS subtab for outgoing subscription messages
4. Submit, when the result returns after a couple of seconds, this message should only come up once:

`["SUBSCRIBE\nid:sub-1567178070746-179\ndestination:/topic/participation/7/newResults\n\n\u0000"]`

### Screenshots
